node/src/signature_mngr.rs

```diff
@@ -48,11 +59,79 @@ struct SignatureManager {
 struct SetKey(SK);
 struct Sign(Vec<u8>);
 
+/// Auxiliary methods for SignatureManager actor
+impl SignatureManager {
+    /// Method to persist the extended secret key into storage
+    fn persist_extended_sk(&self, ctx: &mut Context<Self>, extended_sk: ExtendedSK) {
```

Comment:
**(2)** I'm against having actor-related methods inside type methods. IMHO this method shouldn't exist and instead generate the appropriate key in `node.rs`, persist it, and tell the signature manager about it (just as we communicate the config to load to the storage manager)